src/NUnitCommon/nunit.agent.core/AgentOptions.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using NUnit.Engine;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace NUnit.Agents
+{
+    /// <summary>
+    /// All agents, either built-in or pluggable, must be able to 
+    /// handle the options defined in this class. In some cases,
+    /// it may be permissible to ignore them but they should never
+    /// give rise to an error.
+    /// </summary>
+    public class AgentOptions
+    {
+        static readonly char[] DELIMS = new[] { '=', ':' };
+        // Dictionary containing valid options with bool value true if a value is required.
+        static readonly Dictionary<string, bool> VALID_OPTIONS = new Dictionary<string, bool>();
+
+        static AgentOptions()
+        {
+            VALID_OPTIONS["agentId"] = true;
+            VALID_OPTIONS["agencyUrl"] = true;
+            VALID_OPTIONS["debug-agent"] = false;
+            VALID_OPTIONS["debug-tests"] = false;
+            VALID_OPTIONS["trace"] = true;
+            VALID_OPTIONS["pid"] = true;
+            VALID_OPTIONS["work"] = true;
```

Comment:
question: The class-level comment indicates these arguments must be "well-known" to other consumers. Should we consider exposing the option names publicly in any way (not necessarily in this PR). Maybe something like this, but not necessarily exactly like this:
```
public class AgentOptions {
  public static class AgentOptionNames {
    public static readonly string AgentId = "agentId";
  }
}
```

Something like that, if even with private visibility, could be used internally by the class when we parse the values in the if/else block further down as well.

Reply:
That could be needed at some point. If so, it can sit alongside the constants we have for packages. Let's think about  this when launchers are introduced.